docs/client-sdk/javascript/tutorials/build-an-xmtp-hello-world-app.md

```diff
@@ -0,0 +1,578 @@
+     // highlight-start
+     const [messages, setMessages] = useState<DecodedMessage[]>([]);
+     const [client, setClient] = useState<any>();
+     const [xmtpClientAddress, setXmtpClientAddress] = useState<any>();
+     // highlight-end
+
+     // highlight-start
+     const initXmtp = async function () {
+       const signer = await fetchSigner();
+       const xmtp = await Client.create(signer, { env: "production" });
+       const conversation = await xmtp.conversations.newConversation(PEER_ADDRESS);
+       const messages = await conversation.messages({
+         direction: SortDirection.SORT_DIRECTION_DESCENDING,
+       });
+
+       setClient(conversation);
+       setMessages(messages);
+       setXmtpClientAddress(xmtp.address);
+       // highlight-end
+     };
+     // highlight-start
+     useEffect(() => {
+       if (xmtpClientAddress) {
+         const streamMessages = async () => {
+           const newStream = await client.streamMessages();
+           for await (const msg of newStream) {
+             setMessages((prevMessages) => {
+               const messages = [...prevMessages];
+               messages.unshift(msg);
+               return messages;
+             });
+           }
+         };
+         streamMessages();
+       }
+     }, [client, xmtpClientAddress]);
+     // highlight-end
+
+     // highlight-start
+     const onSendMessage = async () => {
+       const message = "gm XMTP bot!";
+       await client.send(message);
+     };
+    // highlight-end
+
+     // highlight-start
+     const MessageList: React.FC<MessageListProps> = ({ msg }) => {
+       return (
+         <ul>
+           {msg.map((message, index) => (
+             <li key={index}>{message.content}</li>
+           ))}
+         </ul>
+       );
+     };
+     // highlight-end
+
+     return (
+       <div className="App">
+         <Web3Button />
+         <Web3Modal
+           projectId="<YOUR_PROJECT_ID>"
+           ethereumClient={ethereumClient}
+         />
+         // highlight-start
+         {isConnected && xmtpClientAddress && (
+           <>
+             <MessageList msg={messages} />
+             <button onClick={onSendMessage}>Send GM</button>
+           </>
+         )}
+         {isConnected && !xmtpClientAddress && (
+           <button onClick={initXmtp}>Connect to XMTP</button>
+         )}
+         // highlight-end
+       </div>
+     );
+   }
+
+   export default App;
+
+   ```
+   When you run the code above, it should give you a blank skeleton like the picture below:
+   ![barebone bot app](img/gm-bot-bare.png)
+
+
+   ## Step 5. Let's add some styles and make it look pretty!
+   Okay, so now that we have all the functionality working, we are going to use Tailwind to pretty it up.
+
+   :::note
+   Please remember to switch out `<YOUR_PROJECT_ID>` with a real `YOUR_PROJECT_ID` number that you created on [WalletConnect Cloud](https://cloud.walletconnect.com/sign-in).  
+   :::
```

Comment:
The formatting seems to be off - the callout isn't displaying in the rendered HTML: https://junk-range-possible-git-create-hw-tutorial-v2-xmtp-labs.vercel.app/docs/client-sdk/javascript/tutorials/build-an-xmtp-hello-world-app#step-5-lets-add-some-styles-and-make-it-look-pretty

I would try removing the extra spaces before the `:::note`, `Please...`, and `:::` - might fix it?

Reply:
Thanks for catching this! I tried running eveyrthing through prettier and i guess that broke it. Weird